### PR TITLE
boost:update dependencies

### PR DESCRIPTION
**Pull request by mathbunnyru**
Specify library name and version:  **boost/all**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/boost/all/conanfile.py [hunk 1]
```diff
@@ -141,7 +141,7 @@ def configure(self):
                     raise ConanInvalidConfiguration("Boost '%s' library requires multi threading" % lib)
 
     def build_requirements(self):
-        self.build_requires("b2/4.2.0")
+        self.build_requires("b2/4.3.0")
```
Comment (SpaceIm) [inline recipes/boost/all/conanfile.py:144]: minor change: could you move `build_requirements` method between `package_id` and `source` methods? (to follow `conan install`'s executions order. It makes recipes easier to understand... at least for me :p )

Comment (mathbunnyru) [reply recipes/boost/all/conanfile.py]: Sure, no problems

**Comment by mathbunnyru** [comment]
I will merge master, it already helped me once

**Review by danimtb — APPROVED** [review]

**Comment by uilianries** [comment]
@mathbunnyru Please, consider https://github.com/mathbunnyru/conan-center-index/pull/2

**Comment by mathbunnyru** [comment]
Thanks, @uilianries !
Merge you PR, and merge master.

**Comment by uilianries** [comment]
> b2: relocation error: b2: symbol _ZNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEE10_M_replaceEmmPKcm, version GLIBCXX_3.4.21 not defined in file libstdc++.so.6 with link time reference

b2 uses gcc5, so the glibc version is incompatible.

**Comment by mathbunnyru** [comment]
Could you please help me to fix this?

**Comment by danimtb** [comment]
@mathbunnyru nothing you can do from your side. This is a known problem we are trying to mitigate #213. We will try to push the fixes soon

**Comment by mathbunnyru** [comment]
There might be a chance that this issue was resolved after some changed in https://github.com/conan-io/conan-center-index/issues/213

Updates some libs and merged master to check the theory.

**Comment by mathbunnyru** [comment]
@uilianries will it work, if I don't update b2 version?

**Review by danimtb — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**